docs/instructlab-deepsearch-integration.md

```diff
@@ -0,0 +1,52 @@
+
+# DeepSearch + InstructLab Integration Proposal
+
+<https://github.com/DS4SD>
+
+## Why is a Conversion Tool Necessary?
+
+Managing submissions for the open-source InstructLab project has revealed a significant bottleneck in processing
+knowledge documents. For the InstructLab backend to effectively utilize these documents, they must be in markdown
+format. Currently, we only accept Wikipedia articles, but the built-in conversion tool is inadequate. Internally at
+IBM, and other companies, many knowledge submissions are in multiple document formats, including PDF format,
+necessitating conversion to markdown before being used in InstructLab.
+
+Existing open-source methods, such as PanDoc, are inconsistent. While they preserve text, they struggle with parsing
+tables and special symbols, as evidenced by issues in PR #1154 of the taxonomy repo in the InstructLab project. Other
+open-source solutions have similar shortcomings.
+
+## Why DeepSearch?
+
+IBM's DeepSearch software excels in document conversion, outperforming traditional open-source methods. Utilizing a
+computer vision model layer, it accurately parses content in the files, including titles, headers, and tables.
+Additionally, it automatically implements RAG layers for models, which could benefit the InstructLab process in
+the future.
+
+## Integration Proposal
+
+To maintain the open-source nature of the project while leveraging the strengths of DeepSearch, we propose a
+two-pronged approach:
+
+### Open-Source Conversion
+
+- Implement a basic document conversion tool in the UI using an open-source method such as PanDoc. This tool will be
+lightweight and easily hosted, ensuring it can be used and improved by the community.
```

Comment:
```suggestion
- Open source conversion: Implement a basic document conversion tool in the InstructLab UI using an open source tool such as [Pandoc](https://pandoc.org/). This tool will be lightweight and easily hosted, ensuring it can be used and improved by the community.
```